src/Xamarin.Android.Build.Tasks/Utilities/ResourceIdentifier.cs

```diff
@@ -42,5 +43,20 @@ public static string CreateValidIdentifier (string identifier)
 
 			return result;
 		}
+
+		internal static string GetResourceName (string type, string name, Dictionary<string, string> map, TaskLoggingHelper log)
+		{
+			string mappedValue;
+			string key = string.Format ("{0}{1}{2}", type, Path.DirectorySeparatorChar, name).ToLowerInvariant ();
```

Comment:
Is there a way we can know if `map` is already ignoring the case, then we can skip a `ToLowerInvariant()`?